astropy/nddata/nduncertainty.py

```diff
@@ -358,7 +376,204 @@ def _propagate_divide(self, other_uncert, result_data, correlation):
+                # If the uncertainty has a different unit than the result we
+                # need to convert it to the results unit.
+                this = to_variance(self.array * self.unit).to(result_data.unit**2).value
+            else:
+                this = to_variance(self.array)
+        else:
+            this = 0
+
+        # Formula: sigma**2 = dA + dB +/- 2*cor*sqrt(dA*dB)
+        # Formula: sigma**2 = sigma_other + sigma_self +/- 2*cor*sqrt(dA*dB)
+        #     (sign depends on whether addition or subtraction)
+
+        # Determine the result depending on the correlation
+        if isinstance(correlation, np.ndarray) or correlation != 0:
+            corr = 2 * correlation * np.sqrt(this * other)
+            result = this + other + correlation_sign * corr
+        else:
+            result = this + other
+
+        return from_variance(result)
+
+    def _propagate_multiply_divide(self, other_uncert, result_data,
+                                   correlation,
+                                   divide=False,
+                                   to_variance=lambda x: x,
+                                   from_variance=lambda x: x):
+        """
+        Error propagation for multiplication or division of variance or
+        variance-like uncertainties. Uncertainties are calculated using the
+        formulae for variance but can be used for uncertainty convertible to
+        a variance.
+
+        Parameters
+        ----------
+
+        other_uncert : `~astropy.nddata.NDUncertainty` instance
+            The uncertainty, if any, of the other operand.
+
+        result_data : `~astropy.nddata.NDData` instance
+            The results of the operation on the data.
+
+        correlation : float or `numpy.ndarray`-like
+            Correlation of the uncertainties.
+
+        divide : bool, optional
+            If ``True``, propagate for division, otherwise propagate for
+            multiplication.
+
+        to_variance : function, optional
+            Function that will transform the input uncertainties to variance.
+            The default assumes the uncertainty is the variance.
+
+        from_variance : function, optional
+            Function that will convert from variance to the input uncertainty.
+            The default assumes the uncertainty is the variance.
+        """
+        # For multiplication we don't need the result as quantity
+        if isinstance(result_data, Quantity):
+            result_data = result_data.value
+
+        if divide:
+            correlation_sign = -1
+        else:
+            correlation_sign = 1
+
+        if other_uncert.array is not None:
+            # We want the result to have a unit consistent with the parent, so
+            # we only need to convert the unit of the other uncertainty if it
+            # is different from its data's unit.
+            if (other_uncert.unit and
+                to_variance(1 * other_uncert.unit) !=
```

Comment:
Why the `1 *` here and in the lines below?

Reply:
I believe this is to force them to be quantities...

Reply:
Ah, thanks!